internal/chain/create_data_account.go

```diff
@@ -20,5 +21,57 @@ func (CreateDataAccount) Validate(st *StateManager, tx *transactions.GenTransact
 		return fmt.Errorf("invalid payload: %v", err)
 	}
 
-	return errors.New("not implemented") // TODO
+	dataAccountUrl, err := url.Parse(body.Url)
+	if err != nil {
+		return fmt.Errorf("invalid account URL: %v", err)
+	}
+
+	//only the ADI can create the data account associated with the ADI
+	if !dataAccountUrl.Identity().Equal(st.SponsorUrl) {
+		return fmt.Errorf("%q cannot sponsor %q", st.SponsorUrl, dataAccountUrl)
+	}
+
+	//if we have a manger book URL, then we need to parse it.
+	var managerBookUrl string
+	if len(body.ManagerKeyBookUrl) != 0 {
+		u, err := url.Parse(body.ManagerKeyBookUrl)
+		if err != nil {
+			return fmt.Errorf("manager key book specified, but url is invalid: %v", err)
+		}
+		managerBookUrl = u.String()
+	}
```

Comment:
Is it OK for the manager key book to belong to a different ADI?

Reply:
Yes